stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/NonMemberChatEventHandler.kt

```diff
@@ -0,0 +1,108 @@
+package io.getstream.chat.android.offline.querychannels
+
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.api.models.FilterObject
+import io.getstream.chat.android.client.events.ChannelUpdatedByUserEvent
+import io.getstream.chat.android.client.events.ChannelUpdatedEvent
+import io.getstream.chat.android.client.events.MemberAddedEvent
+import io.getstream.chat.android.client.events.MemberRemovedEvent
+import io.getstream.chat.android.client.events.NotificationAddedToChannelEvent
+import io.getstream.chat.android.client.events.NotificationMessageNewEvent
+import io.getstream.chat.android.client.events.NotificationRemovedFromChannelEvent
+import io.getstream.chat.android.client.models.Channel
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.runBlocking
+
+/**
+ * Implementation of [ChatEventHandler] that handles events when ChannelListViewModel has a filter to show channels which
+ * the current user is not a member. BE AWARE that this implementation uses much more API calls than [DefaultChatEventHandler]
+ */
+public class NonMemberChatEventHandler(
+    private val client: ChatClient,
+    private val channels: StateFlow<List<Channel>>,
+) : BaseChatEventHandler() {
+
+    override fun handleNotificationAddedToChannelEvent(
+        event: NotificationAddedToChannelEvent,
+        filter: FilterObject,
+    ): EventHandlingResult = EventHandlingResult.Skip
+
+    override fun handleMemberAddedEvent(
+        event: MemberAddedEvent,
+        filter: FilterObject,
+    ): EventHandlingResult = handleMemberUpdate(event.cid, filter)
```

Comment:
Do we need to do a request here? Shouldn't we just add a channel to the list?

Reply:
We actually just need to remove it (the logic is inverse here).

Reply:
We need to make a request for Channel because we there's no Channel in the event `MemberRemovedEvent`

Reply:
Our API to add channels doesn't work with cid, only the whole channel